client/lobby/RandomMapTab.cpp

```diff
@@ -205,60 +169,153 @@ void RandomMapTab::updateMapInfoByHost()
-	group->setSelected(randIndex);
+	mapGenOptions = opts;
+	
+	//prepare allowed options
+	for(int i = 0; i <= PlayerColor::PLAYER_LIMIT_I; ++i)
+	{
+		playerCountAllowed.insert(i);
+		compCountAllowed.insert(i);
+		playerTeamsAllowed.insert(i);
+		compTeamsAllowed.insert(i);
+	}
+	auto * tmpl = mapGenOptions->getMapTemplate();
+	if(tmpl)
+	{
+		playerCountAllowed = tmpl->getPlayers().getNumbers();
+		compCountAllowed = tmpl->getCpuPlayers().getNumbers();
+	}
+	if(mapGenOptions->getPlayerCount() != CMapGenOptions::RANDOM_SIZE)
+	{
+		vstd::erase_if(compCountAllowed,
+		[opts](int el){
+			return PlayerColor::PLAYER_LIMIT_I - opts->getPlayerCount() < el;
+		});
+		vstd::erase_if(playerTeamsAllowed,
+		[opts](int el){
+			return opts->getPlayerCount() <= el;
+		});
+		
+		if(!playerTeamsAllowed.count(opts->getTeamCount()))
+		   opts->setTeamCount(CMapGenOptions::RANDOM_SIZE);
+	}
+	if(mapGenOptions->getCompOnlyPlayerCount() != CMapGenOptions::RANDOM_SIZE)
+	{
+		vstd::erase_if(playerCountAllowed,
+		[opts](int el){
+			return PlayerColor::PLAYER_LIMIT_I - opts->getCompOnlyPlayerCount() < el;
+		});
+		vstd::erase_if(compTeamsAllowed,
+		[opts](int el){
+			return opts->getCompOnlyPlayerCount() <= el;
+		});
+		
+		if(!compTeamsAllowed.count(opts->getCompOnlyTeamCount()))
+			opts->setCompOnlyTeamCount(CMapGenOptions::RANDOM_SIZE);
+	}
+	
+	if(auto w = widget<CToggleGroup>("groupMapSize"))
+		w->setSelected(vstd::find_pos(getPossibleMapSizes(), opts->getWidth()));
+	if(auto w = widget<CToggleButton>("buttonTwoLevels"))
+		w->setSelected(opts->getHasTwoLevels());
+	if(auto w = widget<CToggleGroup>("groupMaxPlayers"))
+	{
+		w->setSelected(opts->getPlayerCount());
+		deactivateButtonsFrom(*w, playerCountAllowed);
+	}
+	if(auto w = widget<CToggleGroup>("groupMaxTeams"))
+	{
+		w->setSelected(opts->getTeamCount());
+		deactivateButtonsFrom(*w, playerTeamsAllowed);
+	}
+	if(auto w = widget<CToggleGroup>("groupCompOnlyPlayers"))
+	{
+		w->setSelected(opts->getCompOnlyPlayerCount());
+		deactivateButtonsFrom(*w, compCountAllowed);
+	}
+	if(auto w = widget<CToggleGroup>("groupCompOnlyTeams"))
+	{
+		w->setSelected(opts->getCompOnlyTeamCount());
+		deactivateButtonsFrom(*w, compTeamsAllowed);
+	}
+	if(auto w = widget<CToggleGroup>("groupWaterContent"))
+	{
+		w->setSelected(opts->getWaterContent());
+		if(opts->getMapTemplate())
+		{
+			std::set<int> allowedWater(opts->getMapTemplate()->getWaterContentAllowed().begin(), opts->getMapTemplate()->getWaterContentAllowed().end());
+			deactivateButtonsFrom(*w, allowedWater);
+		}
+		else
+			deactivateButtonsFrom(*w, {-1});
+	}
+	if(auto w = widget<CToggleGroup>("groupMonsterStrength"))
+		w->setSelected(opts->getMonsterStrength());
+	if(auto w = widget<CButton>("templateButton"))
+	{
+		if(tmpl)
+			w->addTextOverlay(tmpl->getName(), EFonts::FONT_SMALL);
+		else
+			w->addTextOverlay("default", EFonts::FONT_SMALL);
```

Comment:
Not 100% sure, but shouldn't this string be translatable?
Also, if "default" means "template will be selected automatically" then I think it should say "Random"

Reply:
> shouldn't this string be translatable?
Should be
> it should say "Random"
I took all of this from HDMod to have exactly as there. Anyway will make it configurable